src/CMakeLists.txt

```diff
@@ -420,6 +438,16 @@ add_library(Halide
             # Including these as sources works around the need to "install" Halide_initmod
             $<TARGET_OBJECTS:Halide_initmod>
             $<TARGET_OBJECTS:Halide_c_templates>)
+if (BUILD_SERIALIZATION)
+    find_package(flatbuffers REQUIRED)
+    if (flatbuffers_FOUND)
+        target_link_libraries(Halide PRIVATE flatbuffers)
+        message(STATUS "Build with flatbuffers")
+    else ()
+        message(FATAL_ERROR "flatbuffers not found")
+    endif ()
+endif ()
```

Comment:
The code above defining the custom command should be deleted. Here are the changes I made:

1. Generate in `${PROJECT_BINARY_DIR}/include`. _n.b._ test the install target; if the generated header is installed, then `packaging/CMakeLists.txt` will need to be updated to exclude it.
2. Don't check `flatbuffers_FOUND` since `REQUIRED` was passed.
3. Just attach the header to the `Halide` target. There was no `.c`/`.cpp` file to include in the "object" library anyway.
4. Only bother with `$<SHELL_PATH:...>` if it's actually necessary in practice.

```suggestion
if (BUILD_SERIALIZATION)
    find_package(flatbuffers REQUIRED)
    target_link_libraries(Halide PRIVATE flatbuffers::flatbuffers)
    
    set(fb_def "${CMAKE_CURRENT_SOURCE_DIR}/halide_ir.fbs")
    set(fb_header "${PROJECT_BINARY_DIR}/include/halide_ir_generated.h")
    add_custom_command(
        OUTPUT "${fb_header}"
        COMMAND flatbuffers::flatc --cpp -o "${PROJECT_BINARY_DIR}/include" "${fb_def}"
        DEPENDS "${fb_def}"
        VERBATIM
    )
    target_sources(Halide PRIVATE "${fb_header}")
endif ()
```

Generating it in `${PROJECT_BINARY_DIR}/include` directory will likely require an additional rule to exclude it from the installation archive.